src/Processors/Processor.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace Processors;
+
+public abstract class Processor : BaseThread
+{
+    private bool _isPaused = false;
+    private bool _onPauseOnce = false;
+    private bool _onResumeOnce = false;
+    private bool _stopRequested = false;
+    private int _interval = 500;
+    public ProcessorStatus status = ProcessorStatus.None;
+
+    protected Processor()
+    {
+        OnCreate();
+    }
+
+    public virtual void OnCreate() => status = ProcessorStatus.Created;
+    public virtual void OnStart() => status = ProcessorStatus.Starting;
+    public virtual void OnRun() => status = ProcessorStatus.Running;
+    public virtual void OnPause() => status = ProcessorStatus.Paused;
+    public virtual void OnResume() => status = ProcessorStatus.Running;
+    public virtual void OnStop() => status = ProcessorStatus.Stopped;
+    public abstract void Process(int interval);
+    public override void RunThread()
+    {
+        OnStart();
+
+        while (!_stopRequested)
+        {
+            if (!_isPaused)
+            {
+                if (!_onResumeOnce && _onPauseOnce)
+                {
+                    OnResume();
+                }
+                else if (!_onPauseOnce)
+                {
+                    OnRun();
+                }
+                _onPauseOnce = false;
+
+                Process(_interval);
+            }
+            else
+            {
+                if (!_onPauseOnce)
+                {
+                    OnPause();
+                    _onPauseOnce = true;
+                }
+                _onResumeOnce = false;
+            }
+        }
+
+        OnStop();
+    }
+
+    public void PauseRequest() => _isPaused = true;

```

Comment:
I don't think you need any of these. Except by interval

Reply:
Taking into account what I previously explained, please let me know if you have advice to improve the approach to this problem.

Reply:
I reworked this class. Please check the updated version.